### PR TITLE
Add tests for lambda

### PR DESCRIPTION
**Pull request by Nyarum**
Work for #86 

Run tests via Makefile (For example):
- `make test_lambda`

Output, like:
```
go test -v ./cmd/lambda
=== RUN   TestLambdaCreateFunction
Step 1 : FROM iron/lambda-nodejs
 ---> 5c05d7c44eae
Step 2 : ADD test.js ./test.js
 ---> Using cache
 ---> 676d1aa4f13e
Step 3 : CMD runtime.function
 ---> Using cache
 ---> 25dcabae16ba
Successfully built 25dcabae16ba
--- PASS: TestLambdaCreateFunction (0.17s)
=== RUN   TestLambdaTestFunction
--- PASS: TestLambdaTestFunction (0.99s)
=== RUN   TestLambdaPublishFunction
--- PASS: TestLambdaPublishFunction (1.02s)

PASS   github.com/iron-io/ironcli/cmd/lambda   3.231s
```

### CODE REVIEW

**Comment by Nyarum** [comment]
@nikhilm still in progress.

**Comment by Nyarum** [comment]
@nikhilm R4R.

File: circle.yml [hunk 1]
```diff
@@ -31,9 +31,12 @@ dependencies:
 
 test:
   pre:
+    - echo $TEST_DOCKER_HUB_CREDS > $HOME/.dockercfg
```
Comment (nikhilm) [inline circle.yml]: Is this defined in circle config? Why can't we just reuse the credentials used by #95 and have the tests log themselves in?

File: cmd/lambda/lambda_test.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	lambdaCreateFunction := NewLambdaCreateFunction()
+	lambdaCreateFunction.FunctionName = "irontesting/test"
+	lambdaCreateFunction.Runtime = "nodejs"
+	lambdaCreateFunction.Handler = "runtime.function"
+	lambdaCreateFunction.FileNames = []string{"../../tests/test.js"}
+
+	err := lambdaCreateFunction.Action()
+	if err != nil {
+		t.Error(err)
+	}
```
Comment (nikhilm) [inline cmd/lambda/lambda_test.go]: Could we please have tests that tested the error cases, and not just the success cases. Is the intention to do this in followups?

**Comment by nikhilm** [comment]
Please don't use all these labels for a project so tiny. They don't help and just require more clicks all over the place.

File: circle.yml (hunk 1 above)
Comment (Nyarum) [inline circle.yml]: @nikhilm yep, this is defined. It related to how lambda package reading a docker configuration - https://github.com/iron-io/lambda/blob/master/lambda/lambda.go#L406

File: cmd/lambda/lambda_test.go (hunk 2 above)
Comment (Nyarum) [inline cmd/lambda/lambda_test.go]: @nikhilm of course, this is base tests for sure that we have working functions.

**Pull request merged**